src/resources/ble_darwin.mm

```diff
@@ -738,12 +749,27 @@ - (id)initWithResource:(toit::BleResource*)resource {
   return proxy;
 }
 
-PRIMITIVE(create_central_manager) {
+PRIMITIVE(create_adapter) {
   ARGS(BleResourceGroup, group);
 
   ByteArray* proxy = process->object_heap()->allocate_proxy();
   if (proxy == null) FAIL(ALLOCATION_FAILED);
 
+  BleAdapterResource* adapter_resource = _new BleAdapterResource(group);
+  group->register_resource(adapter_resource);
+
+  proxy->set_external_address(adapter_resource);
+  return proxy;
+}
+
+PRIMITIVE(create_central_manager) {
+  ARGS(BleAdapterResource, adapter);
+
+  auto group = adapter->group();
+
+  ByteArray* proxy = process->object_heap()->allocate_proxy();
+  if (proxy == null) FAIL(ALLOCATION_FAILED);
+
   BleCentralManagerResource* central_manager_resource = _new BleCentralManagerResource(group);
```

Comment:
Maybe add a comment that explains that his cannot fail?

Reply:
done.